pkg/controllers/node/emptiness.go

```diff
@@ -86,7 +86,8 @@ func (r *Emptiness) isEmpty(ctx context.Context, n *v1.Node) (bool, error) {
 	if err := r.kubeClient.List(ctx, pods, client.MatchingFields{"spec.nodeName": n.Name}); err != nil {
 		return false, fmt.Errorf("listing pods for node %s, %w", n.Name, err)
 	}
-	for _, p := range pods.Items {
+	for i := range pods.Items {
```

Comment:
I'm seeing this pattern in a lot of places now. Is this a lint recommendation? I'm not sure I agree that this is better. Is this supposed to protect against the pointer location changing out from under the local loop var?

Reply:
Only when it's closed over or its address is taken.